airflow/models.py

```diff
@@ -2485,7 +2485,10 @@ def __lt__(self, other):
     def __hash__(self):
         hash_components = [type(self)]
         for c in self._comps:
-            val = getattr(self, c, None)
+            if c == 'task_ids':
```

Comment:
Definitely on board for that. Ironically, the addition of the task_ids property was from when we were trying to replace the *first* speed regression in 1.7.1 and I think it was supposed to mimic the old behavior (which stored task_ids in a list) for compatibility. Since this is a blocker, I vote let's not make the change here but maybe for 1.8.